File: 3783.mirror-distance-of-an-integer/solution.kt
class Solution {
    fun mirrorDistance(n: Int): Int {
        return abs(n - reverse(n))
    }

    fun reverse(n: Int): Int {
        var r = 0
        var k = n
        while (k > 0) {
            r = 10 * r + k % 10
            k /= 10
        }
        return r
    }
}

File: 3783.mirror-distance-of-an-integer/solution.go
func mirrorDistance(n int) int {
    return abs(n - reverse(n))    
}

func abs(n int) int {
    if n < 0 {
        return -n
    }
    return n
}

func reverse(n int) int {
    res := 0
    for ; n > 0; n /= 10 {
        res = 10*res + n%10
    }
    return res
}